src/routing_membrane.rs

```diff
@@ -502,7 +496,8 @@ impl<F> RoutingMembrane<F> where F: Interface {
         }
 
         // Forward
-        ignore(self.send_swarm_or_parallel_or_relay(&message));
+        ignore(self.send_swarm_or_parallel_or_relay_with_signature(
+            &message, message_wrap.signature().clone()));
```

Comment:
Could this new `send_swarm_or_parallel_or_relay_with_signature` function just take `SignedMessage` as argument? In this case the `message_wrap` variable.